set_test.go

```diff
@@ -44,3 +44,71 @@ key: [1, "hi"]
 	// key: [1, "hi"]
 	// new: 1
 }
+
+func ExampleSetScalar() {
+	// Load YAML data
+	docNode := new(goyaml.Node)
+	_ = goyaml.Unmarshal([]byte(`---
+# Comment that should be preserved.
+key: [1, "hi"]
+h:
+- abc
+- 123
+`), docNode)
+	node := docNode.Content[0]
+
+	// Edit.
+	_, _ = SetScalar(node, MustParsePath(".new"), "some content")
```

Comment:
Add two more test cases:
- 1 for if a value already exists at the specified path, but has a different value (assert changed=true).
- another for if a value already exists at the specified path, but has equal value (assert changed=false).